internal/core/labels_test.go

```diff
@@ -0,0 +1,37 @@
+package core
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMergeCustomLabels(t *testing.T) {
+	t.Run("merge success", func(t *testing.T) {
+		// --- Given ---
+		dst := map[string]string{"A": "1", "B": "2"}
+		src := map[string]string{"B": "X", "C": "3"}
+
+		// --- When ---
+		err := MergeCustomLabels(dst, src)
+
+		// --- Then ---
+		require.NoError(t, err)
+		assert.Equal(t, map[string]string{"A": "1", "B": "X", "C": "3"}, dst)
```

Comment:
suggestion: Lets always use require so we don't suffer from knock on errors and keep it all consistent, more below
```suggestion
		require.Equal(t, map[string]string{"A": "1", "B": "X", "C": "3"}, dst)
```